packages/connect-agreement/src/entities/Signer.ts

```diff
@@ -0,0 +1,38 @@
+import { SubscriptionHandler } from '@aragon/connect-types'
+
+import Signature from '../entities/Signature'
+import { SignerData, IAgreementConnector } from '../types'
+
+export default class Signer {
+  #connector: IAgreementConnector
+
+  readonly id!: string
+  readonly address!: string
+  readonly agreementId!: string
+
+  constructor(data: SignerData, connector: IAgreementConnector) {
+    this.#connector = connector
+
+    this.id = data.id
+    this.address = data.address
+    this.agreementId = data.agreementId
+  }
+
+  async disconnect() {
+    await this.#connector.disconnect()
+  }
+
+  async hasSigned(versionNumber: string): Promise<boolean> {
```

Comment:
Leaving a comment here to remind ourselves of a good use-case of further data transformation that we can do at the Connector layer.